sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -279,23 +279,20 @@ protected Query applyArrayReference(Reference arrayReference, Literal literal, C
             protected Query applyArrayLiteral(Reference reference, Literal arrayLiteral, Context context) throws IOException {
                 String columnName = reference.ident().columnIdent().fqn();
 
-                if (reference.valueType().equals(DataTypes.STRING)) {
-                    Object values = arrayLiteral.value();
-                    TermsFilter termsFilter;
-                    if (values instanceof Collection) {
-                        termsFilter = new TermsFilter(columnName, getBytesRefs((Collection) arrayLiteral.value()));
-                    } else {
-                        termsFilter = new TermsFilter(columnName, getBytesRefs((Object[]) arrayLiteral.value()));
-                    }
-                    return new FilteredQuery(Queries.newMatchAllQuery(), termsFilter);
-                } else {
-                    QueryBuilderHelper builder = QueryBuilderHelper.forType(reference.valueType());
-                    BooleanFilter filter = new BooleanFilter();
-                    for (Object value : toIterable(arrayLiteral.value())) {
-                        filter.add(builder.eqFilter(columnName, value), BooleanClause.Occur.SHOULD);
-                    }
-                    return new FilteredQuery(Queries.newMatchAllQuery(), filter);
+                Object values = arrayLiteral.value();
```

Comment:
it would probably still make sense to use the query version if there are not too much terms?